pkg/handlers/handlers.go

```diff
@@ -351,7 +361,7 @@ func HandleFile(
 // handleChunksWithError processes data and errors received from the dataErrChan channel.
 // For each DataOrErr received:
 // - If it contains data, the function creates a chunk based on chunkSkel and reports it through the reporter.
-// - If it contains an error, the function logs the error.
+// - If it contains an error, the function returns the error immediately.
```

Comment:
This is for _fatal_ errors, right? This comment seems like a good place to explain non-fatal errors as well. I would also like in some comment _somewhere_ a description of what types of errors are fatal versus non-fatal - maybe in the doc comments of functions that use `DataOrErr`?